### PR TITLE
feat: repo-list can scan multiple orgs; fix file not in current pwd

### PR DESCRIPTION
**Pull request by amenocal**
## Description

This pull request introduces comprehensive support for standalone repository list processing ("repo-list mode") in addition to existing organization-based modes. It adds robust validation, parsing, and execution for repo-list input, ensures mutually exclusive source modes, and includes extensive test coverage for these new features. The documentation is updated to clearly explain usage, constraints, and output differences for repo-list mode.

🚀 : `--repo-list` can now run standalone of `--org-list` and `--org-name` so users can pull files from different organizations in one run.

🐛 : Also fixes a bug where it was expecting the file to be in the `gh extension` directory:

```
src/index.ts → dist/index.js...
created dist/index.js in 3.3s
◇ injected env (0) from .env // tip: ◈ secrets for agents [www.dotenvx.com]
file:///home/gitadmin/.local/share/gh/extensions/gh-repo-stats-plus/dist/index.js:1165
        throw new Error(`File not found: ${filePath}`);
              ^

Error: File not found: repos.txt
    at Option.parseFileAsNewlineSeparatedOption [as parseArg]
 ```

## Checklist

- [x] Issue linked if existing
- [x] Add a label to the pull request that indicates the type of change (e.g., `major`, `minor`, `patch`, `enhancement`, `bug`)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests

## Additional Context

**Repo-list Mode Feature and Validation**
* Added support for processing a standalone file of repositories in strict `owner/repo` format via the `--repo-list` option, including internal grouping by owner for efficient API calls and a single combined CSV/state output. (`README.md`, `repo-list.js`, `main.js`, [[1]](diffhunk://#diff-956cf3592a3e7dd926b02bef64379a71323a2225de9d2348983b025d6a6c73abL2-R11) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R126-R141) [[3]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R349-R351) [[4]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L410-R434)
* Enforced mutual exclusivity between `--repo-list`, `--org-name`, and `--org-list` source modes, with detailed validation and error messages for invalid combinations or empty repo-list input. (`repo-stats-command.js`, `__tests__/commands.test.ts`, [__tests__/commands.test.tsR178-R269](diffhunk://#diff-956cf3592a3e7dd926b02bef64379a71323a2225de9d2348983b025d6a6c73abR178-R269))

**Testing and Robustness**
* Added extensive tests for repo-list parsing, deduplication, grouping, and file path resolution, including handling of malformed entries, comments, whitespace, and relative/absolute paths. (`__tests__/repo-list.test.ts`, [__tests__/repo-list.test.tsR1-R243](diffhunk://#diff-8f82b0969cd46adb49c8c7db7957c5b00c3c06fc43c6fdb865465cead87b5122R1-R243))
* Added integration tests for end-to-end repo-list execution, covering multi-owner runs, state management, rate limit handling, GitHub App auth edge cases, and auto-processing of missing repos. (`__tests__/repo-list-execution.test.ts`, [__tests__/repo-list-execution.test.tsR1-R301](diffhunk://#diff-857931c8dc60a5a2d6a53c518f691244490fd41263f4dc88321e9f07edfd9bfaR1-R301))

**Documentation Updates**
* Updated the `README.md` to document the new `--repo-list` mode, its usage, mutual exclusivity with other source options, state file naming, and behavior for missing repo detection. (`README.md`, [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R126-R141) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R230-R236) [[3]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R349-R351) [[4]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L410-R434)

**CLI and Option Handling**
* Clarified CLI help for source and batch options, distinguishing between standalone `--repo-list` and batched organization runs, and improved error messages for invalid combinations. (`README.md`, [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R377) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L345)

These changes collectively make the tool more flexible and robust for users who need to process an arbitrary set of repositories, not just those belonging to a single organization.

### CODE REVIEW

File: __tests__/wrapper.test.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  it('exports the invocation directory before changing to the extension directory', () => {
+    const wrapper = readFileSync(resolve('gh-repo-stats-plus'), 'utf-8');
+    const exportIndex = wrapper.indexOf(
+      'export GH_REPO_STATS_PLUS_INVOCATION_DIR="$PWD"',
+    );
+    const cdIndex = wrapper.indexOf('cd "$SCRIPT_DIR"');
+
+    expect(exportIndex).toBeGreaterThan(-1);
+    expect(cdIndex).toBeGreaterThan(-1);
+    expect(exportIndex).toBeLessThan(cdIndex);
```
[the remark this answers is not in the document]
Comment (amenocal) [reply __tests__/wrapper.test.ts:15]: fixed in [f96ccc6c2961dd4e0f43023aa8e2484d16643125](https://github.com/mona-actions/gh-repo-stats-plus/commit/f96ccc6c2961dd4e0f43023aa8e2484d16643125)

File: src/commands/repo-stats-command.ts [hunk 2]
```diff
@@ -214,7 +232,7 @@ repoStatsCommand
       'Path to file containing list of repositories to process (format: owner/repo_name)',
     )
       .env('REPO_LIST')
-      .argParser(parseFileAsNewlineSeparatedOption),
+      .argParser(parseRepoListFileOption),
   )
```
[the remark this answers is not in the document]
Comment (amenocal) [reply src/commands/repo-stats-command.ts:229]: fixed in [f96ccc6c2961dd4e0f43023aa8e2484d16643125](https://github.com/mona-actions/gh-repo-stats-plus/commit/f96ccc6c2961dd4e0f43023aa8e2484d16643125)

File: src/main.ts [hunk 3]
```diff
@@ -71,6 +84,431 @@ export async function run(opts: Arguments): Promise<string[]> {
...
+  mode: 'org' | 'repo-list' = 'org',
+): Set<string> {
+  if (mode === 'repo-list') {
+    return new Set(
+      processedState.processedRepos.map((repoName) => repoName.toLowerCase()),
+    );
+  }
+
+  return new Set(
+    processedState.processedRepos.map((repoName) => repoName.toLowerCase()),
```
[the remark this answers is not in the document]
Comment (amenocal) [reply src/main.ts]: fixed in [f96ccc6c2961dd4e0f43023aa8e2484d16643125](https://github.com/mona-actions/gh-repo-stats-plus/commit/f96ccc6c2961dd4e0f43023aa8e2484d16643125)

File: src/main.ts [hunk 4]
```diff
@@ -71,6 +85,400 @@ export async function run(opts: Arguments): Promise<string[]> {
...
+  return `repo-list-all_repos-${timestamp}_ts.csv`;
+}
+
+function usesInstallationAuth(opts: Arguments): boolean {
+  return Boolean(
+    opts.appInstallationId || process.env.GITHUB_APP_INSTALLATION_ID,
+  );
+}
+
+function hasTokenAuth(opts: Arguments): boolean {
```
Comment (scottluskcis) [inline src/main.ts]: auth related stuff should go to auth.ts

File: src/main.ts [hunk 5]
```diff
@@ -71,6 +85,400 @@ export async function run(opts: Arguments): Promise<string[]> {
...
+  if (normalizedRepoList.duplicates.length > 0) {
+    logger.warn(
+      `Ignored ${normalizedRepoList.duplicates.length} duplicate repo-list entr${
+        normalizedRepoList.duplicates.length === 1 ? 'y' : 'ies'
+      }`,
+    );
+  }
+
+  const outputDir = opts.outputDir || 'output';
+  const stateManager = new StateManager(outputDir, 'repo-list', logger);
```
Comment (scottluskcis) [inline src/main.ts]: initializing things is part of init.ts and like we do in the `run` method we build up a context and execute the command is the pattern. `main.ts` has grown 400+ lines with this commit 

```
  const context = await initCommand(opts, config);
  const result = await executeCommand(context, config);
```

File: src/main.ts [hunk 6]
```diff
@@ -71,6 +85,400 @@ export async function run(opts: Arguments): Promise<string[]> {
...
+  }
+
+  if (usesInstallationAuth(opts) && normalizedRepoList.summary.ownerCount > 1) {
+    throw new Error(
+      'Standalone --repo-list with GitHub App installation auth currently supports one owner per run. Use a GitHub token or split the list by owner.',
+    );
+  }
+}
+
+async function runRepoListOnly(opts: Arguments): Promise<string[]> {
```
Comment (scottluskcis) [inline src/main.ts]: it seems like anything that is specific to `runRepoListOnly` might benefit from being in the `repo-list.ts` file?

File: src/main.ts [hunk 7]
```diff
@@ -54,6 +64,10 @@ const repoStatsConfig: CommandConfig = {
 // --- Public entry point ---
 
 export async function run(opts: Arguments): Promise<string[]> {
+  if (isStandaloneRepoListMode(opts)) {
```
Comment (scottluskcis) [inline src/main.ts]: this feels like this check could have been in `repo-stats-command.ts` and `repo-list.ts` house all of the logic we rolled into `main.ts` here that was new.

File: src/main.ts (hunk 6 above)
Comment (scottluskcis) [reply src/main.ts]: I would suggest create a `repo-list-service.ts` or something like that and move the following methods to it 

move 
- runRepoListOnly
- processRepoListOwnerGroup
- processRepositoryByName
- processMissingRepoListRepositories
- findMissingRepoListEntries,
- buildProcessedRepoKeySet
-  and related helpers out of main.ts

File: src/main.ts (hunk 5 above)
Comment (scottluskcis) [reply src/main.ts]: Reuse initCommand/executeCommand infrastructure. extend CommandConfig or create a SourceStrategy interface so repo-list mode uses the same orchestration skeleton (state init, retry setup, session tracking) instead of duplicating it.

**Review by scottluskcis — CHANGES_REQUESTED** [review]
The feature intent is sound but the implementation bypasses the existing architecture and introduces structural debt. Extracting the repo-list execution path into its own module and reusing the existing orchestration infrastructure would bring this in line with the SOLID principles the codebase already follows.

File: src/main.ts (hunk 6 above)
Comment (amenocal) [reply src/main.ts]: @scottluskcis I actually debated about this to create like a separate file for the service, but I didn't see anything like this in the project, so I decide it to skip it 😅

**Review by scottluskcis — APPROVED** [review]

**Review by lannathompson65-arch — COMMENTED** [review]
Great
